CairoMakie/src/primitives.jl

```diff
@@ -928,12 +946,12 @@ function draw_atomic(scene::Scene, screen::Screen, @nospecialize(primitive::Maki
 end
 
 function surface2mesh(xs, ys, zs::AbstractMatrix)
-    ps = Makie.matrix_grid(p-> nan2zero.(p), xs, ys, zs)
+    ps = Makie.matrix_grid(identity, xs, ys, zs)
     rect = Tesselation(Rect2f(0, 0, 1, 1), size(zs))
     faces = decompose(QuadFace{Int}, rect)
     uv = map(x-> Vec2f(1f0 - x[2], 1f0 - x[1]), decompose_uv(rect))
-    uvm = GeometryBasics.Mesh(GeometryBasics.meta(ps; uv=uv), faces)
-    return GeometryBasics.normal_mesh(uvm)
+    uvm = GeometryBasics.Mesh(GeometryBasics.meta(ps; uv=uv), faces, )
```

Comment:
Just do here:
```julia
return Mesh(meta(ps; uv=uv, normals=nan_aware_normals(ps, faces), faces)
```
And skip the whole `nan_aware_*(geometry_func)` business ;)

Reply:
Huh, didn't know you could do that...but should have guessed 😅 

Do we want this kind of behaviour for meshes as well though?  If so we can force recompute the normals in mesh as well.

Reply:
That'd be a pretty big change... I'll just have the `nan_aware_normals` function in Makie, us it for a while and go from there ;)

Reply:
Sounds good.